### ISSUE TITLE
Animation improvements (retargeting)

### ISSUE THREAD

**Issue by Popov72** [BabylonJS/Babylon.js#14963]
Add helpers to simplify animation retargetting.

**Comment by afrokick** [comment]
Related post about Blending for VAT https://forum.babylonjs.com/t/enable-animations-blending-for-vat/49117/3

**Comment by ertugrulcetin** [comment]
Is this ` Implement a layered animation system, "a la" Unity` still in the pipeline? @Popov72

**Comment by Popov72** [comment]
Yes it is :)
We still have a number of things to do, but everything's going well.

**Comment by ertugrulcetin** [comment]
Once it's done, I guess we will have some something similar like this? https://developer.playcanvas.com/user-manual/animation/anim-layer-masking/

**Comment by Popov72** [comment]
Yes, it should be pretty close to what you can find in Unity's mecanim system (layers (additive or not), state machines, different types of blend trees, etc).

**Comment by ertugrulcetin** [comment]
Thank you for the info!

**Comment by Popov72** [comment]
I create a separate issue for baked vertex animation performance:

https://github.com/BabylonJS/Babylon.js/issues/15806

I've also done away with the “Unity-style” layered system, as we'd need an editor to make it a useful addition to the library, which would require a lot of work, especially as the editor would be very different from our existing editors (nme, nge, etc), so we wouldn't be able to reuse our shared code.

**Comment by ertugrulcetin** [comment]
@Popov72 Any updates on Animation retargeting? It'd be great to have this feature, currently it requires so much work to adapt different animations to mixamo models.

**Comment by Popov72** [comment]
No update yet, I plan to take a look at it before our next release, but we have other more important priorities to deal with first...

**Comment by yuripourre** [comment]
This feature would be really nice to have.

**Comment by ltropf** [comment]
Our team has also been keeping an eye on this, as otherwise we have a **lot** of manual work to do for a feature we're wanting to build. 🙏

**Comment by Popov72** [comment]
I'm currently working on it!

**Comment by Popov72** [comment]
Animation retargeting is now available, see:

https://forum.babylonjs.com/t/introducing-animation-retargeting/62547
